pg_search/src/index/search.rs

```diff
@@ -213,9 +217,14 @@ impl SearchIndex {
     /// be entirely owned by the new process, with no references.
     pub fn writer(directory: &WriterDirectory) -> Result<IndexWriter, SearchIndexError> {
         let search_index: Self = directory.load_index()?;
-        let index_writer = search_index
-            .underlying_index
-            .writer(INDEX_TANTIVY_MEMORY_BUDGET)?;
+
+        let memory_budget = if search_index.memory_budget < 15_000_000 {
+            INDEX_TANTIVY_MEMORY_BUDGET_DEFAULT
+        } else {
+            search_index.memory_budget
+        };
```

Comment:
I think we should make the `15_000_000` a const and make sure we have doc comments to explain the number.